config/config.exs

```diff
@@ -62,10 +63,22 @@ rule_evaluator_provider =
 
 config :oli, :rule_evaluator,
   dispatcher: rule_evaluator_provider,
-  node_js_pool_size: String.to_integer(System.get_env("NODE_JS_POOL_SIZE", "2")),
   aws_fn_name: System.get_env("EVAL_LAMBDA_FN_NAME", "rules"),
   aws_region: System.get_env("EVAL_LAMBDA_REGION", "us-east-1")
 
+variable_substitution_provider =
+  case System.get_env("VARIABLE_SUBSTITUTION_PROVIDER") do
```

Comment:
We probably don't need all this logic in the compile-time `config.exs`. We could simplify here by just setting `variable_substitution_provider` to its default, `Oli.Activities.Transformers.VariableSubstitution.NoOpImpl` and remove the `config :oli, :variable_substitution`. As long as the tests and NoOpImpl doesn't expect these configs to exist, this should cover the default configuration and test configuration, while still allowing a runtime configuration to be provided via `releases.exs`

Reply:
I'm not sure I understand what to do here.   I want to ensure that a developer has the ability switch provider, without having the touch any Elixir code.